### PR TITLE
S3/feature/information store in conversations

### PR DESCRIPTION
**Pull request by stu-greenshoots**
Adds a BaseStore that implements defaults for the methods required by the interface
Adds functionality to get supported information requests
adds information_response to the webchat conversation controller to be used in messages
some fixes for warnings

I've put in a couple of fixes to reduce warnings as well and will try to do so with each PR to actsmart

### CODE REVIEW

File: src/Agent.php [hunk 1]
```diff
@@ -237,8 +242,31 @@ public function performAction($action_id, Map $arguments = null)
                 return $this->getActuator($actuator)->perform($action_id, $arguments);
             }
         }
+
+        $this->logger->info(sprintf('No supporting Store found for information request %', $action_id));
```
Comment (istos) [inline src/Agent.php]: I am guessing this want to say "No supporting actuator" ?

File: src/Agent.php [hunk 2]
```diff
@@ -237,8 +242,31 @@ public function performAction($action_id, Map $arguments = null)
...
+    }
+
+    /**
+     * Loops through all available information requests and if they support the requested
+     *
+     * @param $information_request_id
+     * @param Map $utterance
+     * @return mixed|null
+     */
+    public function performInformationRequest($information_request_id, Map $utterance)
```
Comment (istos) [inline src/Agent.php]: An information request might come from something other than an utterance. Still sensible to standardise on a Map data type but don't need to call it utterance here - maybe just arguments?

File: src/Controllers/WebChat/ConversationController.php [hunk 3]
```diff
@@ -62,19 +64,22 @@ public function handleNewConversation(Map $utterance)
...
+        if ($informationRequest = $ci->getCurrentInformationRequest()) {
+            $informationResponse = $this->getAgent()->performInformationRequest($informationRequest, $utterance);
+        }
+
         /* @var \actsmart\actsmart\Conversations\Utterance $nextUtterance */
         $nextUtterance = $ci->getNextUtterance($this->getAgent(), $utterance, $intent, false);
 
         $this->getAgent()->getActuator('actuator.webchat')->perform('action.webchat.postmessage', [
-            Literals::MESSAGE => $nextUtterance->getMessage()->getWebChatResponse($actionResult ?? $utterance)
+            Literals::MESSAGE => $nextUtterance->getMessage()->getWebChatResponse($actionResult ?? $utterance, $informationResponse)
```
Comment (istos) [inline src/Controllers/WebChat/ConversationController.php]: Thinking we need to structure the getWebChatResponse parameter more (potentially an object from which you can extract an action result, or an information response or an utterance). The message "template" can then look for the information it needs  and deal with the cases where it doesn't find it.

File: src/Conversations/Conversation.php [hunk 4]
```diff
@@ -214,11 +219,11 @@ public function getAllUtterancesForScene($scene_id)
     }
 
     /**
-     * @return \Fhaculty\Graph\Vertex
+     * @return Scene
```
Comment (istos) [inline src/Conversations/Conversation.php]: 👏

File: src/Controllers/WebChat/ConversationController.php (hunk 3 above)
Comment (stu-greenshoots) [reply src/Controllers/WebChat/ConversationController.php]: Yes, I agree. And I think that this is something that can be standardised across all message types (Slack, Facebook etc)

**Pull request merged**